routes/transaction.go

```diff
@@ -2548,6 +2547,287 @@ func (fes *APIServer) TransferDAOCoin(ww http.ResponseWriter, req *http.Request)
 	}
 }
 
+// DAOCoinLimitOrderResponse ...
+type DAOCoinLimitOrderResponse struct {
+	SpendAmountNanos  uint64
+	TotalInputNanos   uint64
+	ChangeAmountNanos uint64
+	FeeNanos          uint64
+	Transaction       *lib.MsgDeSoTxn
+	TransactionHex    string
+	TxnHashHex        string
+}
+
+// DAOCoinLimitOrderOperation Use this to represent any operations that can be performed on
+// a DAO coin limit order. For now, we'll only support creating and cancelling order
+type DAOCoinLimitOrderOperation string
+
+const (
+	Create DAOCoinLimitOrderOperation = "CREATE"
+	Cancel DAOCoinLimitOrderOperation = "CANCEL"
+)
+
+// CreateOrCancelDAOCoinLimitOrderRequest ...
+type CreateOrCancelDAOCoinLimitOrderRequest struct {
+	// The public key of the user who is sending the order
+	TransactorPublicKeyBase58Check string `safeForLogging:"true"`
+
+	// Create or Cancel this order
+	Operation DAOCoinLimitOrderOperation
+
+	// The public key or profile username of the DAO coin being bought
+	BuyingDAOCoinCreatorPublicKeyBase58CheckOrUsername string `safeForLogging:"true"`
+
+	// The public key or profile username of the DAO coin being sold
+	SellingDAOCoinCreatorPublicKeyBase58CheckOrUsername string `safeForLogging:"true"`
+
+	// Caller has the option of populating one of these two fields, depending on if they want
+	// to submit this param as a float or want to scale it themselves first. Populating both should
+	// result in an error response
+	ScaledExchangeRateCoinsToSellPerCoinToBuy *uint256.Int `safeForLogging:"true"`
+	ExchangeRateCoinsToSellPerCoinToBuy       float64      `safeForLogging:"true"`
```

Comment:
will a float64 be big enough?

Reply:
Discussed this offline. Yeah, float64 has a wide enough range